packages/cli/src/commands/proxy.ts

```diff
@@ -0,0 +1,46 @@
+import { getHttpOperationsFromResource } from '@stoplight/prism-http';
+import { pick } from 'lodash';
+import { CommandModule } from 'yargs';
+import { createMultiProcessPrism, CreateProxyServerOptions, createSingleProcessPrism } from '../util/createServer';
+import sharedOptions from './sharedOptions';
+
+const proxyCommand: CommandModule = {
+  describe: 'Start a proxy server with the given document file',
+  command: 'proxy <document> <upstream>',
+  builder: yargs =>
+    yargs
+      .positional('document', {
+        description: 'Path to a document file. Can be both a file or a fetchable resource on the web.',
+        type: 'string',
+      })
+      .positional('upstream', {
+        description: 'Url to a target server.',
+        type: 'string',
+      })
+      .coerce('upstream', value => {
+        try {
+          return new URL(value);
+        } catch (e) {
+          throw new Error(`Invalid upstream URL provided: ${value}`);
+        }
+      })
+      .middleware(async argv => (argv.operations = await getHttpOperationsFromResource(argv.document!)))
+      .options(sharedOptions),
+  handler: parsedArgs => {
```

Comment:
`handler` could also be shared

Reply:
it really could ;) so that we have the same logic for `proxy` and `mock`. But maybe we don't want to make this `handler` shared? I'm thinking, do we want `proxy` to also be restarted (hot-reloaded) when the doc is being changed? If so, we might just make `handler` a common functionality. And this is assuming that the hot-reloading will indeed take place here, of which I am still unsure

Reply:
For now, it doesn't look like sth worth sharing. Effectively `pick(...)` varies depending on the command. L42 is one and only common thing. But, according to what you say about hot-reload it may be valuable to share it when there will be more code.

Reply:
This will be shared and refactored — but I need #689 merged before working on this.